### ISSUE TITLE
Custom getUser method

### ISSUE THREAD

**Issue by clem983** [lexik/LexikJWTAuthenticationBundle#848]
Hi,

I'm developing a SASS app and in my database, email addresses are not unique. I added a "company" field: the same account can exist for many companies.

I was thinking I need to rewrite the 'getUser' method (by extending the JWTTokenAuthenticator) in order to take into account the company ID in the query.
But before that, I tried to put some "dumps" in the JWTTokenAuthenticator base file, and the supports() method returns false because there is no bearer token in the request. So the other methods (getCredentials, getUser, ...) are not called.

So, this file is only called to authenticate users from bearer ? But where and how can I customize the method to retrieve a user during login ?

I don't understand that ... could someone help me ? :)

Thank you very much :)

**Comment by chalasr** [comment]
> So, this file is only called to authenticate users from bearer ?

Yes. The JWTTokenAuthenticator is about authenticating from a token previously obtained.

>  But where and how can I customize the method to retrieve a user during login ?

The login part (the step that delivers a token) is managed by Symfony, not this bundle. As such, it can be any Symfony built-in security listener/authenticator. 
The only thing that this bundles does in the login process is to deliver a token upon successful login. 
This is achieved by configuring the `success_handler` and `failure_handler` on your `form_login` (or `json_login` or whatever authentication method used to obtain a token), as shown in the bundle documentation.

> I'm developing a SASS app and in my database, email addresses are not unique. I added a "company" field: the same account can exist for many companies.

You probably need a custom user provider: https://symfony.com/doc/4.0/security/entity_provider.html#using-a-custom-query-to-load-the-user

**Comment by clem983** [comment]
I had paused this feature, but after a few months ... it worked with the custom user provider! Thank you very much ! :)
